src/Accounts/Authentication/AzureSessionInitializer.cs

```diff
@@ -254,18 +254,25 @@ public override SourceLevels AuthenticationTraceSourceLevel
             /// <summary>
             /// Adal Logger for Adal 3.x +
             /// </summary>
-            public AdalLogger AdalLogger { get; private set; }  
+            public AdalLogger AdalLogger { get; private set; }
 
             /// <summary>
             /// Write messages to the existing trace listeners when log messages occur
             /// </summary>
             /// <param name="message"></param>
             private void WriteToTraceListeners(string message)
             {
-                foreach (var listener in AuthenticationTraceListeners)
+                for (var i = 0; i < AuthenticationTraceListeners.Count; ++i) // don't use foreach, enumerator is not thread safe
                 {
-                    var trace = listener as TraceListener;
-                    trace.WriteLine(message);
+                    try
+                    {
+                        var trace = AuthenticationTraceListeners[i] as TraceListener;
```

Comment:
`as TraceListener` is not required.

Reply:
Good point. Removed type conversion and combined two lines together. Thanks